CDCgov/prime-reportstream
# Jessica/7355 record hl7 batch files

This PR ...

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
Using the Postman collection, send three different tests via the Send HCA HL7 Full ELR endpoint found in the full-ELR folder.

Send a singular message without headers and check that it gets marked as HL7 in the body_format field in the report_file table in the db
Add the headers to the message and make sure it gets marked as HL7_BATCH
Remove the headers and add a second message to make sure it gets marked as HL7_BATCH


## Changes
- Detecting HL7 Batch vs HL7
- Saving the report bodyFormat rather than the sender format to the report_file body_format field

## Checklist

### Testing
- [ Y] Tested locally?
- [ Y] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ N/A] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ Y] Added tests?

### Process
- [ N] Are there licensing issues with any new dependencies introduced?
- [ Y] Includes a summary of what a code reviewer should test/verify?
- [ N] Updated the release notes?
- [ N/A] Database changes are submitted as a separate PR?
- [ N/A] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue 7355

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints? No
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints? No
- Does this change introduce new dependencies that need vetting?No 
- Does this change require changes to our infrastructure? No
- Does logging contain sensitive data? No
- Does this PR include or remove any sensitive information itself? No
